src/apify/scrapy/cache.py

```diff
@@ -0,0 +1,176 @@
+from __future__ import annotations
+
+import gzip
+import io
+import pickle
+import struct
+from logging import getLogger
+from time import time
+from typing import TYPE_CHECKING
+
+from scrapy.http.headers import Headers
+from scrapy.responsetypes import responsetypes
+
+from ._async_thread import AsyncThread
+from apify import Configuration
+from apify.apify_storage_client import ApifyStorageClient
+from apify.storages import KeyValueStore
+
+if TYPE_CHECKING:
+    from scrapy import Request, Spider
+    from scrapy.http.response import Response
+    from scrapy.settings import BaseSettings
+    from scrapy.utils.request import RequestFingerprinterProtocol
+
+logger = getLogger(__name__)
+
+
+class ApifyCacheStorage:
```

Comment:
Isn't this specifically HTTP cache storage? (rather then general cache)

Reply:
The built-in Scrapy cache storages are called `DbmCacheStorage` or `FilesystemCacheStorage`. They live in a module `scrapy.extensions.httpcache`. Maybe I could rename the module to `httpcache`.

Reply:
Addressed in 3a2cc13

Reply:
Makes sense, thanks.